swapnil/CSS/Theory.txt

```diff
@@ -0,0 +1,216 @@
+div#section1 span{color:green}
+div p{color:blue}
+p{color:red}
+                </style>
+        </head>
+        <body>
+                <div id="section1">
+                        <p> this is first paragraph </p>
+                </div>
+        </body>
+</html>
+
+-Blue color there is no selector of span tag.div p<immediate parent> so it will execute first.
+
+Q:In which colour will the text “This is a heading!” be printed for the HTML
+  and CSS shown below? Also explain the specificity rule by which that colour
+  is chosen.
+<html>
+        <head>
+                <style>
+#parent{
+        color:green;
+}
+h2{
+        color:purple;
+}
+                </style>
+        </head>
+        <body>
+                <div id="parent">
+                        <h2>This is heading!</h2>
+                </div>
+        </body>
+</html>
+
+-It will in purple color.
+-id=parent is div id.
+-According to rule <h2> is first priority.
+
+Q:Sometimes designers mention multiple font families in their styles
+  (eg: font-family: arial, helvetica, serif). How does this help?
+-It will look first for arial font.if it not found by browser then it will
+ helvetica and then serif.
+
+Q:Mention one place where you have noticed text decoration being used on
+  websites. Also mention the purpose for which text decoration was used in that  place.
+-http://www.htmldog.com/guides/css/beginner/text/ where menu bar options
+ contains text-decoration:underline is used.
+
+Q:Replace the rbg values in the styles shown below with an actual colour value, such as red, blue, green, yellow, etc, without changing the colour of the span
+
+<html>
+        <head>
+                <style>
+#s1{
+        background:rgb(0,0,255);
+}
+#s2{
+        background:rgb(0%,0%,100%);
+}
+                </style>
+        </head>
+        <body>
+                <span id="s1"> Amber Academy </span>
+                <span id="s2"> Amber Academy </span>
+        </body>
+</html>
+
+-<html>
+        <head>
+                <style>
+#s1{
+        background:red;
+   }
+#s2{
+        background:green;
+   }
+                </style>
+        </head>
+        <body>
+                <span id="s1"> Amber Academy </span>
+                <span id="s2"> Amber Academy </span>
+        </body>
+</html>
+
+Q:Show with the help of a proper box diagram, the difference in the way the
+  span element (with id=s1) will be rendered in both the scenarios shown below:
+Scenario A
+
+<html>
+        <head>
+                <style>
+#s1{
+        margin-left:20px;
+        background:yellow;
+}
+                </style>
+                <body>
+                        <span id="s1"> Amber Academy </span>
+                </body>
+</html>
+
+Scenario B
+
+<html>
+        <head>
+                <style>
+#s1{
+        padding-left:20px;
+        background:yellow;
+}
+                </style>
+                <body>
+                        <span id="s1"> Amber Academy </span>
+                </body>
+</html>
+
```

Comment:
bad css and code indentation 
use Tab(4 spaces)

Reply:
done changes

Reply:
wrong indentation

Reply:
@avinash0070  CSS code is given in question itself

Reply:
line no. 177 wrong identation

Reply:
use tab (4space) for HTML close also